src/main/java/hudson/remoting/Launcher.java

```diff
@@ -324,6 +333,9 @@ public void run() throws Exception {
               jnlpArgs.add("-jar-cache");
               jnlpArgs.add(jarCache.getPath());
             }
+            if (this.disableHttpEndpointCheck) {
```

Comment:
Isn't the `slaveJnlpUrl` in the surrounding `if` an http URL? Does the `disableHttpEndpointCheck` parameter make sense here then?